src/contracts/atlas/GasAccounting.sol

```diff
@@ -237,71 +249,167 @@ abstract contract GasAccounting is SafetyLocks {
     )
         internal
     {
-        // Calculate what the solver owes if they failed
-        // NOTE: This will cause an error if you are simulating with a gasPrice of 0
-        if (!bidFind && !result.updateEscrow()) return;
-
-        uint256 gasUsed = (gasWaterMark - gasleft() + _SOLVER_LOCK_GAS_BUFFER) * tx.gasprice;
+        uint256 gasUsed = (gasWaterMark + _SOLVER_BASE_GAS_USED - gasleft()) * tx.gasprice;
 
         if (includeCalldata) {
             gasUsed += _getCalldataCost(solverOp.data.length);
         }
 
-        gasUsed = (gasUsed + ((gasUsed * SURCHARGE_RATE) / SURCHARGE_SCALE));
-        _assign(solverOp.from, gasUsed, false, bidFind);
+        gasUsed = (gasUsed * ((SURCHARGE_SCALE + ATLAS_SURCHARGE_RATE + BUNDLER_SURCHARGE_RATE) / SURCHARGE_SCALE));
+
+        // Calculate what the solver owes
+        // NOTE: This will cause an error if you are simulating with a gasPrice of 0
+        if (!bidFind && !result.updateEscrow()) {
+            // CASE: Solver is not responsible for the failure of their operation, so we blame the bundler
+            // and reduce the total amount refunded to the bundler
+            writeoffs += gasUsed;
+        } else {
+            // CASE: Solver failed, so we calculate what they owe.
+            uint256 deficit = _assign(solverOp.from, gasUsed, false, bidFind);
+            if (deficit > 0) {
+                // Write off any deficit as a gas loss to the bundler so that other solvers aren't forced to pay it.
+                writeoffs += deficit;
+            }
+        }
+    }
+
+    /// @param ctx Context struct containing relavent context information for the Atlas auction.
+    /// @param solverGasLimit The maximum gas limit for a solver, as set in the DAppConfig
+    function _getAdjustedClaimsAndWriteoffs(
+        Context memory ctx,
+        uint256 solverGasLimit
+    )
+        internal
+        view
+        returns (uint256 _claims, uint256 _writeoffs)
+    {
+        _claims = claims;
+        _writeoffs = writeoffs;
+
+        uint256 _gasLeft = gasleft(); // Hold this constant for the calculations
+
+        uint256 _gasRemainder =
+            _gasLeft * tx.gasprice * (SURCHARGE_SCALE + ATLAS_SURCHARGE_RATE + BUNDLER_SURCHARGE_RATE) / SURCHARGE_SCALE;
+        _claims -= _gasRemainder;
+
+        // By reducing the _claims, Solvers end up paying less in total.
+        if (ctx.solverCount > 0) {
+            uint256 _bundlerGasOveragePenalty;
+
+            // Calculate the unadjusted bundler gas surcharge
```

Comment:
`_bundlerGasOveragePenalty` is unused in this scope
```suggestion
        if (ctx.solverCount > 0) {
            // Calculate the unadjusted bundler gas surcharge
```

Reply:
Yeah looks to me like this one can be removed and it can be declared at the bottom as it already is - the only place this var is used